src/plugins/visualizations/public/vis_types/vis_type_alias_registry.ts

```diff
@@ -30,7 +30,7 @@ export interface VisualizationListItem {
 export interface VisualizationsAppExtension {
   docTypes: string[];
   searchFields?: string[];
-  toListItem: (savedObject: SimpleSavedObject) => VisualizationListItem;
+  toListItem: (savedObject: SimpleSavedObject<any>) => VisualizationListItem;
```

Comment:
I think that we use the `SerializableRecord` type for the attributes, can we use this instead of any?

Reply:
It does not work. I don't think you need to type it as you don't use the attributes. You just communicate that any type of SO is passed so `any` is OK.

<img width="995" alt="Screenshot 2023-04-06 at 12 56 17" src="https://user-images.githubusercontent.com/2854616/230372433-2396f5c3-3435-482e-9797-3b8d514f5da4.png">

Reply:
nit: We should really avoid the use of `any`. It indicates we have no full confidence in our type system.

imho, for this one in particular, plugging in `<any>` is obfuscating what's happening. We're changing the interface from a SO-type now to either accept a CM-type or a SO-type (in this specific case, just only a single `MapItem` implementation).

Given that there's very few implementations of this `toListItem`-method, I would suggest making it very explicit until everything is migrated over. It's really only incompatible because of the light change in `attributes`.

e.g.: consider

```
type ListTypeSource = Pick<SimpleSavedObject, 'id' | 'updatedAt'> & {
  attributes: {
    title: string;
    description?: string;
  };
};

export interface VisualizationsAppExtension {
  docTypes: string[];
  searchFields?: string[];
  toListItem: (savedObject: ListTypeSource) => VisualizationListItem;
}
```
Might also have to fix this stub (https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/src/plugins/visualizations/public/utils/saved_visualize_utils.test.ts#L460)
something like: `(savedObject.attributes as unknown as { label: string }).label`

Reply:
I disagree. `any` has a meaning and using it does not mean we don't have a confidence in our type system.

In this scenario what we are saying is: "_I will pass you a saved Object with **any** attributes, please give me back a `VisualizationListItem`_". 

The way it is implemented is 

```ts
// a saved object with any attributes, maybe it does not even have a "title" or "description"
toListItem(savedObject) {
  // you have the savedObject, now return a VisualizationListItem
  
  // This is the important part
  return {
    id,
    title,
    description,
    updatedAt,
    editUrl: getEditPath(id),
    editApp: APP_ID,
    icon: APP_ICON,
    stage: 'production' as VisualizationStage,
    savedObjectType: type,
    typeTitle: APP_NAME,
  };
},
```

And if we _really_ want to type the full tree, then some generic should be pass elsewhere and then passed back in the `SavedObject<T>`. But I really think it is overkill.

Remember that before my change it was `unknown`

```ts
toListItem(savedObject: SimpleSavedObject) { // it was "unknown"
  const { id, type, updatedAt, attributes } =
    savedObject as SimpleSavedObject<MapSavedObjectAttributes>; // we forced the casting
```